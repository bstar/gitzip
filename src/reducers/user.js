import {
    SET_GITHUB_KEY,
    GET_USER_SUCCESS,
    GET_USER_REPOS_SUCCESS,
    SET_ACTIVE_REPO,
    GET_USER_REPO_ISSUES_SUCCESS,
  } from '../constants/ActionTypes';
  
  const initialSettings = {};
  
  const user = (state = initialSettings, action) => {
    
    switch (action.type) {
  
      case SET_GITHUB_KEY:
    
        const key = action.payload;

        return {
          ...state,
          key,
        }

      case GET_USER_SUCCESS:
  
        const userData = action.payload;

        return {
          ...state,
          login: userData.login,
          avatar_url: userData.avatar_url,
          url: userData.url,
          name: userData.name,
        }

      case GET_USER_REPOS_SUCCESS:

        const repos = action.payload;

        return {
          ...state,
          repos,
        }

      case SET_ACTIVE_REPO:

        const activeRepoData = action.payload;

        return {
          ...state,
          activeRepoData,
          issues: null,
        }

        case GET_USER_REPO_ISSUES_SUCCESS:

          const issues = action.payload;

          return {
            ...state,
            issues,
          }

      default:
        return state;
    }
  };
  
  export default user;
  