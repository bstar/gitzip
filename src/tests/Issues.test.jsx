import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducers from '../reducers';
import Issues from '../components/issues';


const store = createStore(
    reducers,
);

it('renders correctly', () => {
    const tree = renderer.create(<Provider store={store}><Issues /></Provider>).toJSON();

    expect(tree).toMatchSnapshot();
});
