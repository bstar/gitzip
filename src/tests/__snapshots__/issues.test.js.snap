// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="issues-container"
>
  <div
    className="content-header"
  >
    Order issues for: 
  </div>
  <div
    className="warning"
  >
    This repo has no issues.
  </div>
</div>
`;
